lib/dependency_validator.dart

```diff
@@ -108,61 +106,59 @@ void run({
       packagesUsedInPublicFiles.add(match.group(1));
     }
   }
-
   logger.fine('packages used in public facing files:\n'
       '${bulletItems(packagesUsedInPublicFiles)}\n');
 
-  // Recursively list all Dart files in known directories other than lib/
-  final nonLibDartFiles = <File>[]
-    ..addAll(listDartFilesIn('example/', excludedDirs))
-    ..addAll(listDartFilesIn('test/', excludedDirs))
-    ..addAll(listDartFilesIn('tool/', excludedDirs))
-    ..addAll(listDartFilesIn('web/', excludedDirs));
-  final nonLibScssFiles = <File>[]
-    ..addAll(listScssFilesIn('example/', excludedDirs))
-    ..addAll(listScssFilesIn('test/', excludedDirs))
-    ..addAll(listScssFilesIn('tool/', excludedDirs))
-    ..addAll(listScssFilesIn('web/', excludedDirs));
-  final nonLibLessFiles = <File>[]
-    ..addAll(listLessFilesIn('example/', excludedDirs))
-    ..addAll(listLessFilesIn('test/', excludedDirs))
-    ..addAll(listLessFilesIn('tool/', excludedDirs))
-    ..addAll(listLessFilesIn('web/', excludedDirs));
+  final nonPublicDartFiles = listDartFilesIn('./', [
+    ...excludes,
+    for (final dir in publicDirs) Glob('$dir**'),
+  ]);
```

Comment:
[nit] this list of exclude globs here is the same for the next two usages, as well. Could move it to a variable to DRY this up.